### PR TITLE
Add support for custom Swift toolchains

### PR DESCRIPTION
**Pull request by brentleyjones**

### CODE REVIEW

File: xcodeproj/internal/bazel_integration_files/bazel_build.sh [hunk 1]
```diff
@@ -177,6 +186,7 @@ log=$(mktemp)
   build \
   --config=$config \
   --color=yes \
+  ${toolchain:+--define=SWIFT_CUSTOM_TOOLCHAIN="$toolchain"} \
```
Comment (keith) [inline xcodeproj/internal/bazel_integration_files/bazel_build.sh:189]: note that this doesn't really work in a lot of ways last I tried

Comment (brentleyjones) [reply xcodeproj/internal/bazel_integration_files/bazel_build.sh:189]: What breaks?

Comment (brentleyjones) [reply xcodeproj/internal/bazel_integration_files/bazel_build.sh:189]: https://github.com/bazelbuild/rules_swift#building-with-custom-toolchains is front and center, so if it doesn't work 😬

Comment (keith) [reply xcodeproj/internal/bazel_integration_files/bazel_build.sh:189]: linking will probably fail because it doesn't know about the toolchain path

Comment (brentleyjones) [reply xcodeproj/internal/bazel_integration_files/bazel_build.sh:189]: Worked in my, very minimal, testing: https://app.buildbuddy.io/invocation/ae071594-1485-41bb-9b97-675b3d88598d

**Review by thii — APPROVED** [review]

File: xcodeproj/internal/bazel_integration_files/bazel_build.sh (hunk 1 above)
Comment (keith) [reply xcodeproj/internal/bazel_integration_files/bazel_build.sh:192]: At the very least you need `--action_env=TOOLCHAINS=toolchain_id` to make sure that wrapped_clang invocations can use the same toolchain. In some situations you'll also need `--host_action_env` with the same

**Pull request merged**